console_backend/src/tabs/settings_tab.rs

```diff
@@ -620,19 +622,17 @@ impl SettingsTab {
     fn group_changed(&self, conf: &Ini, group: &str) -> Result<bool> {
         let new_group = match conf.section(Some(group)) {
             Some(s) => s,
-            None => return Ok(false),
+            None => return Ok(false), // no change, no need to update
         };
         let guard = self.settings.lock();
-        for SettingsEntry { setting, value } in guard.group(group)? {
-            let new_value = match new_group.get(&setting.name) {
-                Some(s) => s,
-                None => continue,
-            };
-            let old_value = match value.as_ref().map(|v| v.to_string()) {
-                Some(v) => v,
-                None => return Ok(true),
-            };
-            if new_value != old_value {
+        let groups = match guard.get_group(group) {
+            Some(v) => v,
+            None => return Ok(true), // update since current values are empty
+        };
+        for (name, new_value) in new_group.iter() {
```

Comment:
this was basically the one of the main issue.

it did not check ntrip.debug as 'old values' did not have this field.
and hence when we tried to toggle ntrip.debug, we didn't actually disable ntrip which would make the first error